src/hazelcore/semantics/dynamics/Dynamics.re

```diff
@@ -4,72 +4,113 @@ open Sexplib.Std;
  * renumbering is done on the final result (see below) */
 
 module Pat = {
-  type expand_result =
-    | Expands(DHPat.t, HTyp.t, Contexts.t, Delta.t)
-    | DoesNotExpand;
+  module ExpandResult = {
+    type t =
+      | Expands(DHPat.t, HTyp.t, Contexts.t, Delta.t)
+      | DoesNotExpand;
+
+    let to_option =
+      fun
+      | DoesNotExpand => None
+      | Expands(pat, ty, ctx, delta) => Some((pat, ty, ctx, delta));
+
+    let from_option =
+      fun
+      | None => DoesNotExpand
+      | Some((pat, ty, ctx, delta)) => Expands(pat, ty, ctx, delta);
+
+    // for use with ppx_let. Don't know why I need to wrap it in a module twice
+    module Let_syntax = {
```

Comment:
I don't think you're using this. But if you do want to use it, the right way to avoid nesting is to do `module Let_syntax = Pat.ExpandResult.Let_syntax` instead of `open Pat.ExpandResult.Let_syntax`, since ppx_let uses the current binding of `Let_syntax`.

Reply:
see https://github.com/janestreet/ppx_let

Reply:
I think I did what you had in mind. Have a look :)